projenrc/api-extractor-docs-publishing.ts

```diff
@@ -0,0 +1,150 @@
+import type { Monorepo, TypeScriptWorkspace } from 'cdklabs-projen-project-types/lib/yarn';
+import { Component, github } from 'projen';
+
+export interface ApiExtractorDocsPublishingProps {
+  /**
+   * The docs stream to publish to.
+   */
+  readonly docsStream: string;
+
+  /**
+   * The role arn (or github expression) for OIDC to assume to do the actual publishing.
+   */
+  readonly roleToAssume: string;
+
+  /**
+   * The bucket name (or github expression) to publish to.
+   */
+  readonly bucketName: string;
+}
+
+export class ApiExtractorDocsPublishing extends Component {
+  private readonly github: github.GitHub;
+  private readonly props: ApiExtractorDocsPublishingProps;
+
+  constructor(project: TypeScriptWorkspace, props: ApiExtractorDocsPublishingProps) {
+    super(project);
+
+    const gh = (project.parent! as Monorepo).github;
+    if (!gh) {
+      throw new Error('This workspace does not have a GitHub instance');
+    }
+    this.github = gh;
+
+    this.props = props;
+
+    // Add a task to run api-extractor and zip the output
+    const apiExtractorDocsTask = project.addTask('api-extractor-docs', {
+      exec: [
+        // Ensure API Extractor is installed
+        'npm install -g @microsoft/api-extractor || true',
+        // Run api-extractor to generate the API model
+        // Use || true to ensure the task continues even if api-extractor reports failures
+        'api-extractor run --local --diagnostics || true',
+        // Create a directory for the API model
+        'mkdir -p dist/api-extractor-docs/cdk/api/toolkit-lib',
+        // Copy the API model to the directory (with error handling)
+        'if [ -f dist/toolkit-lib.api.json ]; then cp dist/toolkit-lib.api.json dist/api-extractor-docs/cdk/api/toolkit-lib/; else echo "Warning: API JSON file not found"; fi',
+        // Add version file
+        '(cat dist/version.txt || echo "latest") > dist/api-extractor-docs/cdk/api/toolkit-lib/VERSION',
+        // Find and copy all markdown files (excluding node_modules)
```

Comment:
I reckon we can limit this to all files in the `docs` directory and `README.md`

You will probably want _all_ files because we might have images.

Reply:
Got it, that makes sense to me. I was thinking we might have .md files in subdirectories, but if that isn't the case, then we won't need anything outside of docs and README.md. And will include all file types.